### ISSUE TITLE
Partition rebalancing

### ISSUE THREAD

**Issue by intouchkey** [segmentio/kafka-go#674]
**ConsumerRebalance Listener methods implementations in kafka-go**

Is there a way for me to flush and commit offsets on partitions revoked?

**Comment by dmarkhas** [comment]
You need to use the consumergroup API. See an example here:

https://github.com/segmentio/kafka-go/blob/master/example_consumergroup_test.go

**Comment by nlsun** [comment]
@intouchkey Yes, @dmarkhas provided a good example of how to use consumer groups, and there is some additional documentation here on how it works as well https://pkg.go.dev/github.com/segmentio/kafka-go#ConsumerGroup

We don't have support for setting `enable.auto.commit` but we do support similar functionality via explicit commits https://github.com/segmentio/kafka-go#explicit-commits

Let me know if this answers your questions!
